### DISCUSSION TITLE
GridList for a custom Select with Search.

### DISCUSSION THREAD

**Discussion by pavelbelyaev98** [adobe/react-spectrum#7100] [Q&A] [answered]
Hello, I am building a custom single and multi select component that also has a search to it. 
Similar to:
![image](https://github.com/user-attachments/assets/df64cd00-d771-42ee-8f02-4a1565ffb7bb)

For this I am using a DialogTrigger + Popover for the overlays and inside render a GridList.

So far so good, but there seems to be no way to automatically focus on the GridList when I open the modal. I have to click "Tab" in order to make the GridList focusable but i want it to behave similarly to normal select where you open the modal and can directly navigate with up and down arrows.

**Comment by snowystinger** [comment]
Can you use `autoFocus`?
I see it on ListBox https://react-spectrum.adobe.com/react-aria/ListBox.html prop table.
I don't see it on GridList, so possibly it was missed and just needs to be added. I can't think of a reason we wouldn't support it. Hopefully it's just missed in the types

https://github.com/adobe/react-spectrum/blob/d57bd8d906ec8f943285798269bbbfd84d2da3eb/packages/%40react-aria/gridlist/src/useGridList.ts#L32

vs 

https://github.com/adobe/react-spectrum/blob/d57bd8d906ec8f943285798269bbbfd84d2da3eb/packages/%40react-types/listbox/src/index.d.ts#L18

**Reply by pavelbelyaev98** [reply]
Adding autoFocus doesnt seem to work, looks like it is missing from GridList

**Reply by snowystinger** [reply, marked as answer]
Thanks for checking that out. I've opened an issue for it https://github.com/adobe/react-spectrum/issues/7105
Would you like to contribute?
